src/main/java/nextstep/courses/domain/session/SessionCapacity.java

```diff
@@ -0,0 +1,57 @@
+package nextstep.courses.domain.session;
+
+import nextstep.courses.CannotEnrollException;
+
+import java.util.Objects;
+
+public class SessionCapacity {
+    private static final int MIN_CAPACITY = 1;
+
+    private int capacity;
+    private final int maxCapacity;
+
+    public SessionCapacity(int capacity, int maxCapacity) {
+        this.capacity = capacity;
+        this.maxCapacity = maxCapacity;
+        checkValidCapacity();
+    }
+
+    public SessionCapacity(int maxCapacity) {
+        this(0, maxCapacity);
+    }
+
+    private void checkValidCapacity() {
+        if (maxCapacity < MIN_CAPACITY) {
+            throw new IllegalArgumentException();
+        }
+        isValidCapacity();
+    }
+
+    public void increase() {
+        capacity++;
+        isValidCapacity();
+    }
```

Comment:
```suggestion
    public void increase() {
        capacity++;
        isValidCapacity();
    }
```

해당 코드를 보고 의문점이 드는게 한가지 있는데요,

```java
        try {
            SessionCapacity sessionCapacity = new SessionCapacity(10, 5);
            sessionCapacity.increase();
        } catch (CannotEnrollException e) {
            System.out.println(e.getMessage());
        }

        // blabla
```

위 테스트 코드는 `10`어야 할까요 `11`이어야할까요?

그렇다면 무엇이 문제일까요?

```java
public void increase() {
    isValidCapacity();
    capacity++;
}
```

위치를 조정해보거나, `capacity` 를 불변객체로 관점 전환 후 해결해볼수도 있을겄같네요 🤔